doc/strategies/AMMWrapper.md

```diff
@@ -0,0 +1,40 @@
+
+Crrently `AMMWrapperWithPath` support following AMM protocols:
+
+-   Uniswap v2/v3
+-   Sushiswap
+-   Balancer v2
+-   Curve v1/v2 (only part of the pools)
+
+## Order Format
+
+| Field            |  Type   | Description                                                         |
+| ---------------- | :-----: | ------------------------------------------------------------------- |
+| makerAddr        | address | The address of the AMM protocol.                                    |
+| takerAssetAddr   | address | The address of taker token.                                         |
+| makerAssetAddr   | address | The address of maker token.                                         |
+| takerAssetAmount | uint256 | The amount of taker token.                                          |
+| makerAssetAmount | uint256 | The amount of maker token.                                          |
+| userAddr         | address | The address of the user.                                            |
+| receiverAddr     | address | The address of the token receiver. It may differ from user address. |
+| salt             | uint256 | A random number included in an order to avoid replay attack.        |
+| deadline         | uint256 | The timestamp of the expiry.                                        |
+
+## Relayer
+
+Tokenlon provides a relaying service for `AMMWrapper` and `AMMWrapperWithPath` transaction. A user can also choose submitting trasanction by themselves. However, official relayers can adjust fee factor dynamically in order to reflect the network congestion but normal users can't.
+
+## Fee
+
+Some portion of the output from AMM will be deducted as protocol fee. The fee will be transfered to `feeCollector` during the settlement. The dynamic fee factor is obtained by `_feeFactor` parameter but for those self-submitted transactions, it will be replaced by the `defaultFeeFactor` of the contract.
+
+## Raw ETH
+
+Each AMM protocol may support either WETH or ETH. The `AMMWrapperWithPath` contract provides flexibilities when it comes to an order with taker asset or maker asset is ETH or WETH. An user can choose whether ETH or WETH as the input and ouptut for such order. It should be specified in the order and the `AMMWrapperWithPath` contract will handle WETH wrap/unwrap if needed.
+
+# AMMQuoter
+
+The `AMMQuoter` contract is a read-only contract providing quotes from different AMM protocols. An reference quote is required when creating an order since AMM swap is always successful but the price varies. By integrating supported AMM protocols, the `AMMQuoter` can provide quotes in a single contract call which simplify the complexity of order creation and reduce the number of RPC call as well.
```

Comment:
What does this sentence mean? Or what is our expected message reader receive from it?
`An reference quote is required when creating an order since AMM swap is always successful but the price varies.`

Reply:
Swap with AMM will always be successful but the price may not be acceptable by the user. So getting an approx quote is required before user commit it and that's what `AMMQuoter` does.

Reply:
I think "swap with AMM will always be successful" could be misleading. 
`It gets a reference quote which display estimated price to user.` will be more straightforward.

Reply:
@pilagod any thoughts on this?

Reply:
Or maybe we could say "It's highly recommended to get quote from AMMQuoter before placing order to AMM, to ensure the swap price is ideal."

Since user could build arbitrary AMM order without quoting, so maybe it's more like a suggestion for user to get quote first to ensure a not-bad price.

Reply:
Drop the `always successful` part and adjust description in d1f5529.